File: ui/src/types/apiCredential.ts
export interface APICredential {
  id: string
  provider: string
  project_id?: string
  secret_ref: string
  created_at: string
  updated_at: string
}

export interface APICredentialCreateRequest {
  provider: string
  project_id?: string
  secret_ref: string
}

export interface APICredentialUpdateRequest {
  provider?: string
  project_id?: string
  secret_ref?: string
}

File: be/internal/model/api_credential.go
package model

import "time"

// APICredential stores a provider API key reference (env/file/literal).
type APICredential struct {
	ID        string    `json:"id"`
	Provider  string    `json:"provider"`
	ProjectID *string   `json:"project_id,omitempty"`
	SecretRef string    `json:"secret_ref"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}
